### PR TITLE
Fix null pointer when creating ConnectorSession for LocalExchange

### PR DESCRIPTION
**Pull request by kewang1024**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by kewang1024** [comment]
similar fix: https://github.com/prestodb/presto/pull/13975

**Comment by kewang1024** [comment]
Blocks https://github.com/prestodb/presto/pull/14095

File: presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java [hunk 1]
```diff
@@ -173,7 +173,7 @@ private static PartitionFunction createPartitionFunction(
         ConnectorNodePartitioningProvider partitioningProvider = partitioningProviderManager.getPartitioningProvider(partitioning.getConnectorId().get());
         ConnectorBucketNodeMap connectorBucketNodeMap = partitioningProvider.getBucketNodeMap(
                 partitioning.getTransactionHandle().orElse(null),
-                session.toConnectorSession(),
+                session.toConnectorSession(partitioning.getConnectorId().get()),
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java]: why the original code has null pointer issue? Maybe explain it a bit in commit message?

Comment (kewang1024) [reply presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java]: ``toConnectorSession()`` will call
``new FullConnectorSession(this, identity.toConnectorIdentity());``

public FullConnectorSession(Session session, ConnectorIdentity identity)
    {
        this.session = requireNonNull(session, "session is null");
        this.identity = requireNonNull(identity, "identity is null");
        this.properties = null;
        this.connectorId = null;
        this.catalog = null;
        this.sessionPropertyManager = null;
        this.sqlFunctionProperties = session.getSqlFunctionProperties();
    }

here will set properties/connecotId/catalog/sessionPropertyManager null

hence we should call ``ConnectorSession toConnectorSession(ConnectorId connectorId)``
public ConnectorSession toConnectorSession(ConnectorId connectorId)
    {
        requireNonNull(connectorId, "connectorId is null");

        return new FullConnectorSession(
                this,
                identity.toConnectorIdentity(connectorId.getCatalogName()),
                connectorProperties.getOrDefault(connectorId, ImmutableMap.of()),
                connectorId,
                connectorId.getCatalogName(),
                sessionPropertyManager);
    }

**Pull request merged**